docs/dev_configuration_options.rst

```diff
@@ -0,0 +1,45 @@
+
+
+Options
+-------
+
+============== ============ =========================================
+Config         Default      Description
+============== ============ =========================================
+BUILD_TYPE     Release      The type of build Release, Debug etc.
+-------------- ------------ -----------------------------------------
+BUILD_SHARED   ON           Build shared libraries (so, dylib, DLLs).
+-------------- ------------ -----------------------------------------
+TWAE :sup:`*`          ON           Treat warnings as errors
+-------------- ------------ -----------------------------------------
+INSTALL_PREFIX /usr/lib     Install path prefix (platform specific)
+-------------- ------------ -----------------------------------------
+UNIT_TESTS     ON           Enable tests.
+-------------- ------------ -----------------------------------------
+MEMCHECK       ON           Enable memcheck testing. (if available)
+-------------- ------------ -----------------------------------------
+COVERAGE       ON           Enable coverage testing. (if available)
+============== ============ =========================================
+
```

Comment:
Minor stuff: a full stop is used for some descriptions and not for some others. Otherwise, another case where ```BUILD_TYPE``` and ```Release``` (and the like) could be written using ``` ``BUILD_TYPE`` ``` and ``` ``Release`` ```.